### DISCUSSION TITLE
Remotetemp not updating

### DISCUSSION THREAD

**Discussion by chetkelley** [emsesp/EMS-ESP32#1889] [Q&A] [answered]
hello all, I noticed a few days ago that the virtual thermostat I set up at the end of last winter has disappeared from my ems-esp. I created a new one (RC100H) and sent temperature and humidity settings via the web interface. The ems-esp created the thermostat and my Buderus system recognized it. However the values for both sensors are not being updated by my OpenHAB system.  I have added 3 things relating to the system that were autodiscovered by Openhab.  1 - a thermostat with a single channel (sensor/ems-esp/thermostat_hc2_currtemp)  2 - a Boiler. 3 - A system thing with ems-esp info.   I tried using MQTT-Explorer to publish a new temperature to the topic sensor/ems-esp/thermostat_hc2_remotetemp but the value isn't updated in the ems-esp web interface.  Here is my systeminfo:
`{
  "System": {
    "version": "3.7.0-dev.26",
    "uptime": "000+05:40:27.142",
    "uptime (seconds)": 20427,
    "platform": "ESP32",
    "arduino": "Tasmota Arduino v2.0.17",
    "sdk": "4.4.8.240628",
    "free mem": 119,
    "max alloc": 55,
    "free caps": 69,
    "used app": 1690,
    "free app": 4006,
    "partition": "app1",
    "reset reason": "Software reset CPU / Software reset CPU"
  },
  "Network": {
    "network": "WiFi",
    "hostname": "ems-esp",
    "RSSI": -52,
    "BSSID": "set",
    "TxPower setting": 0,
    "static ip config": false,
    "low bandwidth": false,
    "disable sleep": false,
    "enable MDNS": true,
    "enable CORS": false,
    "AP provision mode": "disconnected",
    "AP security": "wpa2",
    "AP ssid": "ems-esp"
  },
  "NTP": {
    "NTP status": "connected",
    "enabled": true,
    "server": "time.google.com",
    "tz label": "Europe/Berlin"
  },
  "MQTT": {
    "MQTT status": "disconnected",
    "enabled": false,
    "client id": "esp32-59e2dec4",
    "keep alive": 60,
    "clean session": false,
    "entity format": 0,
    "base": "ems-esp",
    "discovery prefix": "homeassistant",
    "discovery type": 0,
    "nested format": 1,
    "ha enabled": false,
    "mqtt qos": 0,
    "mqtt retain": false,
    "publish time heartbeat": 60,
    "publish time boiler": 10,
    "publish time thermostat": 10,
    "publish time solar": 10,
    "publish time mixer": 10,
    "publish time water": 10,
    "publish time other": 60,
    "publish time sensor": 10,
    "publish single": false,
    "publish2command": false,
    "send response": false
  },
  "Syslog": {
    "enabled": false
  },
  "Sensor": {
    "temperature sensors": 0,
    "temperature sensor reads": 0,
    "temperature sensor fails": 0,
    "analog sensors": 0,
    "analog sensor reads": 0,
    "analog sensor fails": 0
  },
  "API": {
    "API calls": 2,
    "API fails": 1
  },
  "Bus": {
    "bus status": "connected",
    "bus protocol": "Buderus",
    "bus telegrams received (rx)": 82482,
    "bus reads (tx)": 20249,
    "bus writes (tx)": 3,
    "bus incomplete telegrams": 8,
    "bus reads failed": 0,
    "bus writes failed": 0,
    "bus rx line quality": 100,
    "bus tx line quality": 100
  },
  "Settings": {
    "board profile": "S32",
    "locale": "en",
    "tx mode": 1,
    "ems bus id": 11,
    "shower timer": false,
    "shower min duration": 180,
    "shower alert": false,
    "hide led": false,
    "notoken api": false,
    "readonly mode": false,
    "fahrenheit": false,
    "dallas parasite": false,
    "bool format": 1,
    "bool dashboard": 1,
    "enum format": 1,
    "analog enabled": true,
    "telnet enabled": true,
    "max web log buffer": 50,
    "web log buffer": 50
  },
  "Devices": [
    {
      "type": "boiler",
      "name": "Enviline/Compress 6000AW/Hybrid 3000-7000iAW/SupraEco/Geo 5xx/WLW196i/WSW196i",
      "device id": "0x08",
      "product id": 172,
      "brand": "",
      "version": "02.02",
      "entities": 154,
      "handlers received": "0xBF 0xC2 0xD1 0xE3 0xE4 0xE5 0xE9 0x0494 0x0495 0x048F",
      "handlers fetched": "0x14 0xE6 0xEA 0x048D 0x048A 0x04A2 0x0485 0x0486 0x0492 0x0488 0x0484 0x048B 0x0491 0x0499 0x049C 0x049D 0x02CC",
      "handlers pending": "0x10 0x11 0x15 0x1C 0x18 0x19 0x1A 0x35 0x16 0x33 0x34 0x26 0x2A 0x28 0x04AE 0x04AF 0x04A5",
      "handlers ignored": "0x049F 0x04A0 0x04A3 0x0240 0xE7 0x02E0 0x02EA 0x061E 0xD2 0xE2 0xF7 0x04AA 0x2E 0x3B 0x0497"
    },
    {
      "type": "thermostat",
      "name": "RC300/RC310/Moduline 3000/1010H/CW400/Sense II/HPC410",
      "device id": "0x10",
      "product id": 158,
      "brand": "",
      "version": "73.03",
      "entities": 111,
      "handlers received": "0x06 0x02BB 0x02BC 0x02BD 0x02BE 0x02BF 0x02C0 0x031D 0x0267",
      "handlers fetched": "0x02A5 0x02B9 0x02AF 0x029B 0x0471 0x02A6 0x02BA 0x02B0 0x029C 0x0472 0x02CC 0x0467 0x0291 0x02CE 0x0468 0x0292 0x0293 0x0294 0x02F5 0x023A 0x0240 0xBB 0x023E 0x0269",
      "handlers pending": "0xA3 0xA2 0x12 0x13 0x02A7 0x02B1 0x029D 0x0473 0x02A8 0x02B2 0x029E 0x0474 0x02A9 0x02B3 0x029F 0x0475 0x02AA 0x02B4 0x02A0 0x0476 0x02AB 0x02B5 0x02A1 0x0477 0x02AC 0x02B6 0x02A2 0x0478 0x02D0 0x0469 0x02D2 0x046A 0x02F6 0x031B 0x031E",
      "handlers ignored": "0x0239 0x0508 0x0509 0x050A 0x0507 0xBF 0x042C 0xF6 0xF7"
    },
    {
      "type": "thermostat",
      "name": "RC100H",
      "device id": "0x39",
      "product id": 200,
      "brand": "",
      "version": "40.04",
      "entities": 3,
      "handlers received": "0x042C 0x047C",
      "handlers pending": "0x0274 0x0A6B"
    },
    {
      "type": "mixer",
      "name": "MM100",
      "device id": "0x21",
      "product id": 160,
      "brand": "",
      "version": "24.05",
      "entities": 7,
      "handlers received": "0x02D8",
      "handlers fetched": "0x02CE",
      "handlers ignored": "0x02E2 0x02EC 0xE6 0x0255 0x0292 0xBF"
    },
    {
      "type": "gateway",
      "name": "WiFi module",
      "device id": "0x48",
      "product id": 252,
      "brand": "Buderus",
      "version": "08.01",
      "entities": 0,
      "handlers ignored": "0xF9 0x2040 0x02BA 0x02BB 0x02BC 0x02C3 0x0495 0x0494 0x3B 0xE5 0xE9 0x02C4 0x02C5 0x02C6 0x02E0 0x02F5 0x02FF 0x0449 0x044A 0x044B 0x17 0x044C 0x0593 0x0471 0x0472 0x0473 0x0474 0x0484 0x05A7 0x05A8"
    }
  ]
}`

[emsesp_system_info.txt](https://github.com/user-attachments/files/16361753/emsesp_system_info.txt)

**Comment by proddy** [comment]
what was the exact command you executed and what did the logs show afterwards?

**Comment by chetkelley** [comment]
Using MQTT Explorer, I published "24" to homeassistant/sensor/ems-esp/thermostat_hc2_remotetemp. The log says "published failed, low memory" That line is continuously running through the logs.  I'm kicking myself for not checking the logs earlier. I'll do some homework on this, switch to the stable version and see if that helps.

**Reply by proddy** [reply]
I would stick on 3.7.0 which is about to go stable and has so many more improvements. The low memory is strange, as I see your system has enough available memory and the settings look fine. @MichaelDvP anything you can think of that would cause this flooding?

**Reply by MichaelDvP** [reply]
freemem looks ok, but "max alloc": 55, "free caps": 69 is a bit low, but i also see "MQTT status": "disconnected", "enabled": false,
This can't be the config with publish fail messages.

**Reply by chetkelley** [reply]
I have rebooted the device and it has calmed down. I'm still on 3.7.0-dev26. Using the web interface to set hc2 room temperature from remote to 22 shows this in the logs:

2024-07-24 19:23:06.552 I 82: [command] Calling command thermostat/remotetemp (room temperature from remote) with value 22 and id 2 on device 0x10
2024-07-24 19:23:33.167 I 83: [command] Calling command thermostat/remotehum (room humidity from remote) with value 56 and id 2 on device 0x10

and the 
HC2 RF room temperature sensor entity in the RC100H shows the change. Now I just have to figure out how to do that from OpenHAB.

**Reply by chetkelley** [reply]
BTW, whenever I restart the device the R100H disappears. Its an easy matter to reinstate it.

**Reply by chetkelley** [reply]
After reactivating MQTT and restarting the device, the out of memory log entries are back:

2024-07-24 20:55:12.001 W 4106: [mqtt] Publish failed: low memory

**Reply by chetkelley** [reply]
A factory reset didn't help. After starting MQTT the low memory errors started flooding the logs.

**Reply by proddy** [reply]
Are you using a BBQKees board? 
When that "free caps" gets below 60, MQTT will fail. We'll need to simulate your setup. Is there anything else you're doing differently? I take it you don't have Modbus enabled?

**Reply by chetkelley** [reply]
Yes, I'm using the BBQKees Gateway S32, ems-esp v3.7.0-dev26. I don't have Modbus enabled, and haven't done any customizations. Its a stock setup.

[emsesp_system_info.txt](https://github.com/user-attachments/files/16371461/emsesp_system_info.txt)

**Reply by proddy** [reply]
ok, very strange. I'll set up an S32 and see if I get the same memory consumption

**Reply by MichaelDvP** [reply]
@chetkelley Please don't attach the settings with your passwords. I've removed the file.

154 boiler- and 111 thermostat entities is quite much. Try to save memory by removing unused entities:
Go to customization and and remove all that have no value. 
If there are some others you are not interested in, you can also remove. 
Restart emsesp.

Do you use/need use the mqtt discovery?

**Reply by proddy** [reply]
Michael is right - with the latest dev 3.7.x releases we've added many more entities, which is hitting the limits of the older ESP32 with 4MB. What you could also do is contact BBQKees and see about swaping the S32 to a newer S3 model which has an ESP32-S3 chip and 16MB RAM onboard - you just need to replace the board itself (4 screws).

**Reply by bbqkees** [reply]
With so many entities the problem is in the available RAM on the S32. The S32 uses the standard ESP32 chip which has 520kb of RAM.
The latest S3 and E32 V2 Gateways have an additional 8MB of PSRAM (and 16MB Flash). So about 16 times more RAM available.
They can easily handle the larger amount of entities in their memory.

The S3 has the USB-C port on the side and also the light guide is different than on the S32 so a board swap would also need a new side piece.
If you reply on your order confirmation email or contact me via the contact form on the webshop I can offer a discount on a new S3 or E32 V2 Gateway.

**Reply by chetkelley** [reply]
Thanks bbqkees, I will take you up on your offer.  I've just replied to the confirmation email.

**Comment by chetkelley** [comment]
I didn’t realize those entities are delectable. Once I have the new esp board I will keep that in mind. CheersChetOn Jul 25, 2024, at 12:35, Kees ***@***.***> wrote:﻿
With so many entities the problem is in the available RAM on the S32. The S32 uses the standard ESP32 chip which has 520kb of RAM.
The latest S3 and E32 V2 Gateways have an additional 8MB of PSRAM (and 16MB Flash). So about 16 times more RAM available.
They can easily handle the larger amount of entities in their memory.
The S3 has the USB-C port on the side and also the light guide is different than on the S32 so a board swap would also need a new side piece.
If you reply on your order confirmation email or contact me via the contact form on the webshop I can offer a discount on a new S3 or E32 V2 Gateway.

**Comment by chetkelley** [comment]
The new Gateway arrived today. I have it installed and everything is running splendidly! I have one question though. I was looking at the MQTT topics using MQTT Explorer. There are two topics for the hc2/remotetemp:

homeassistant/number/ems-esp/thermostat_hc2_remotetemp
homeassistant/sensor/ems-esp/thermostat_hc2_remotetemp

Which should I use to publish a temperature from a sensor in my room?

**Reply by MichaelDvP** [reply, marked as answer]
HA topics are retain, you mixed up old and new. Delete allHA-topics  using MQTT expolrer and restart emsesp. Or, in emsesp, disable HA, leave mqtt enabled, wait a few minutes, unntil all HA topics are deleted, enable HA again.

**Comment by chetkelley** [comment]
That did the trick. Thanks for your patience and support.
